src/fsharp/IlxGen.fs

```diff
@@ -4579,6 +4740,33 @@ and GetIlxClosureFreeVars cenv m (thisVars: ValRef list) eenvouter takenNames ex
     // Build the environment that is active inside the closure itself
     let eenvinner = eenvinner |> AddStorageForLocalVals g (thisVars |> List.map (fun v -> (v.Deref, Arg 0)))
 
+    // Work out if the closure captures any witnesses. 
+    let cloWitnessInfos = 
+        let generateWitnesses = ComputeGenerateWitnesses g eenvinner
+        if generateWitnesses then 
+            GetTraitWitnessInfosOfTypars g 0 cloFreeTyvars // TODO: 0 may be wrong here
```

Comment:
I know that, "0", represents the number of parent typars and we are in the process of generating a closure. 

Is this something to worry about now? Could a bug be easily introduced? Is there ever going to be a case where we are generating a closure type inside a generic type?

Reply:
If we can figure out how to make this correct, that would be better and prevent any further issues from cropping up.

Reply:
The 0 here represents that a closure doesn't reside within a generic class - there are no "enclosing class type parameters" to lop off.  

I've removed the TODO and added that comment

Reply:
Great, that makes a lot of sense.